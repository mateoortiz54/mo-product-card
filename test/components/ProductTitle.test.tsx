import React from 'react';

import renderer from 'react-test-renderer'
import ProductCard, {ProductTitle} from '../../src/components'
import { product } from '../data/products';

describe('Pruebas en el ProductTitle', () => {
  test('debe de mostrar el componente correctamente en el titutlo personalizado', () => {
    
    const wrapper = renderer.create(
        <ProductTitle title='Hola mundo'/>    
    )
    
    console.log(wrapper.toJSON())
    expect(wrapper.toJSON()).toMatchSnapshot();

  });

  test('Debe de mostrar el componente con el nombre del producto', () => {
    
    const wrapper = renderer.create(
        <ProductCard product={product}>
            {
                () => (
                    <ProductTitle/>
                )
            }
        </ProductCard>    
    )
    
    expect(wrapper.toJSON()).toMatchSnapshot();


  });    
  

  
});