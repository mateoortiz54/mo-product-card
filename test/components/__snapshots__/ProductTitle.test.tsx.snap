// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el ProductTitle Debe de mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffe-Mug-Card Sin Imagen
  </span>
</div>
`;

exports[`Pruebas en el ProductTitle debe de mostrar el componente correctamente en el titutlo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Hola mundo
</span>
`;
